Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -318,7 +321,32 @@ endRide handle@ServiceHandle {..} rideId req = withLogTag ("rideId-" <> rideId.g
     let mbPaymentUrl = DMPM.getPostpaidPaymentUrl =<< mbPaymentMethod
     let mbPaymentMethodInfo = DMPM.mkPaymentMethodInfo <$> mbPaymentMethod
     withTimeAPI "endRide" "notifyCompleteToBAP" $ notifyCompleteToBAP booking updRide newFareParams mbPaymentMethodInfo mbPaymentUrl
-
+    transporterConfig <- QTConf.findByMerchantOpCityId booking.merchantOperatingCityId >>= fromMaybeM (TransporterConfigNotFound booking.merchantOperatingCityId.getId)
+    timeInIST <- getLocalCurrentTime transporterConfig.timeDiffFromUtc
+    ds <- SQDS.findByDriverIdAndDate driverId (utctDay timeInIST)
+    dailyStatsOfDriver' <- case ds of
+      Nothing -> do
+        id <- generateGUIDText
+        pure
+          DDS.DailyStats
+            { id = id,
+              driverId = driverId,
+              totalEarnings = finalFare,
+              numRides = 1,
+              totalDistance = chargeableDistance,
+              date = utctDay timeInIST
+            }
+      Just dailyStats ->
+        pure
+          DDS.DailyStats
+            { id = dailyStats.id,
+              driverId = driverId,
+              totalEarnings = dailyStats.totalEarnings + finalFare,
+              numRides = dailyStats.numRides + 1,
+              totalDistance = dailyStats.totalDistance + chargeableDistance,
+              date = utctDay timeInIST
+            }
+    SQDS.upsertByDriverId dailyStatsOfDriver'
```

Comment:
in DB keep it as UTC Time instead of IST date and handle timestamps bounds to query for a day in a specific localZones using the timeDIffFromUTC Config.

Reply:
I am storing date in form of "yyyy-mm-dd" and not in complete form of UTCTime. 
From frontend we are sending dates according to LocalTime and in backend if we will store dates in UTC then dates of two different timezones can't be compared accordingly as full UTCTime can't be stored in this case. 
What should I do here ?

Reply:
sure, you are already using `timeDIffFromUTC` config, just name the date column in Db as merchantLocalDate and also don't use IST in any variables you defined above, it gave an impression of being hardcoded to IST.